### ISSUE TITLE
[CRITICAL] Startup fails due to failed HTTPClient Connection Start

### ISSUE THREAD

**Issue by Nicolas-Klatzer** [FrostbyteSpace/openhiven.py#14]
**Summary**
Startup fails due to an unsuccessful start of the connection to Hiven over the HTTPClient. The session can be created but without success in interacting with Hiven. The initial request does not give a correct response and possibly pool dies out while requesting causing `asyncio.TimeoutError` to be thrown.

**Error Place:**
`openhivenpy/gateway/http.py`

**Code:**
```python        
async def _request(endpoint, method, json_data, timeout, **kwargs):
    http_code = "Unknown Internal Error"
    timeout = aiohttp.ClientTimeout(total=timeout)
    if self.http_ready:
        try:
            async with self.session.request(
                                            method=method,
                                            url=f"{self.request_url}{endpoint}", 
                                            headers=self.headers, 
                                            json=json_data, 
                                            timeout=timeout,
                                            **kwargs) as resp: # <- Timeouts because of unsuccessful establishing of connection
                http_code = resp.status
``` 

**Expected behaviour**
The request should proceed without error and continue replacing the empty data of the Client -> Does not continue due to error and crashes the process and keeps the execution_loop as well as the WebSocket loop from starting 

**Possible Cause**
After many attempts of making it work, the bug was discovered to probably lay in some part of the aiohttp or urllib3 module. The asyncio event_loop or nested loop could also possibly a cause for freezing the connection. Was found to be dependent on the system that was used or the packages since no errors were reported on a Linux os(Archlinux x86_64)

**Comment by Nicolas-Klatzer** [comment]
Bug still active in c4af4fbff00ebab6aacdccdb19446ac45d7ac08b

**Comment by Nicolas-Klatzer** [comment]
@Nxybi Can you maybe look into this?

**Comment by Nxybi** [comment]
> @Nxybi Can you maybe look into this?

Believe me - I was straight onto it last night ~~stop taking credit for my bug finding smh~~, and all I've found is it errors in http.py/connect() when trying to super()

**Comment by Nicolas-Klatzer** [comment]
> Believe me - I was straight onto it last night ~stop taking credit for my bug finding smh~, and all I've found is it errors in http.py/connect() when trying to super()

I know that but still, it does not error there because that is just a method that starts the session. In there the error happens when trying to do the initial request. So super() works, just we will need to look deeper into the request itself (Code snippet and reference up there)

**Comment by Nxybi** [comment]
> > Believe me - I was straight onto it last night ~stop taking credit for my bug finding smh~, and all I've found is it errors in http.py/connect() when trying to super()
> 
> I know that but still, it does not error there because that is just a method that starts the session. In there the error happens when trying to do the initial request. So super() works, just we will need to look deeper into the request itself (Code snippet and reference up there)

Kudo.
Do I look that smart?

**Comment by Nicolas-Klatzer** [comment]
@Nxybi Yes you do, those are just excuses not to work xD Simply use breakpoints and inspect where the bad things happen

**Comment by Nxybi** [comment]
> @Nxybi Yes you do, those are just excuses not to work xD Simply use breakpoints and inspect where the bad things happen

I have, kudo, I'm the supervisor since I have jack shit on how to write a fuckin HTTP client

**Comment by Nicolas-Klatzer** [comment]
Oki then I don't really know how we can solve the issue. Especially because it has something to do with the environment where it gets executed and finding there the error will take some time

**Comment by Nxybi** [comment]
> Oki then I don't really know how we can solve the issue. Especially because it has something to do with the environment where it gets executed and finding there the error will take some time

I'll ask around - surely someone knows how to fix

**Comment by Nicolas-Klatzer** [comment]
Okay else I will maybe do a bug report in [aiohttp](https://github.com/aio-libs/aiohttp) because that really doesn't seem like an issue caused by us

**Comment by Nxybi** [comment]
> Okay else I will maybe do a bug report in [aiohttp](https://github.com/aio-libs/aiohttp) because that really doesn't seem like an issue caused by us

I asked one of my friends last night and he gave it the all clear, maybe check aiohttps repo to see if someone there is having the same issue?

**Comment by Nicolas-Klatzer** [comment]
Oh okay, that's nice because I thought I am getting crazy after 6 hours of checking every piece of code and thinking I might have some unseen error and wrote it faulty. The thing is I haven't yet seen anything like it on the issue page. Some stuff on urllib3 and some minor stuff in aiohttp but nothing that really indicated a solution

**Comment by Nxybi** [comment]
> Oh okay, that's nice because I thought I am getting crazy after 6 hours of checking every piece of code and thinking I might have some unseen error and wrote it faulty. The thing is I haven't yet seen anything like it on the issue page. Some stuff on urllib3 and some minor stuff in aiohttp but nothing that really indicated a solution

Nothing on SO either?

**Comment by Nicolas-Klatzer** [comment]
SO?

**Comment by Nxybi** [comment]
> SO?

https://stackoverflow.com/

**Comment by Nicolas-Klatzer** [comment]
oooh okay, yeah didn't find anything. Only this: https://github.com/aio-libs/aiohttp/issues/3203

**Comment by Nxybi** [comment]
> oooh okay, yeah didn't find anything. Only this: [aio-libs/aiohttp#3203](https://github.com/aio-libs/aiohttp/issues/3203)

That is odd, maybe just ask around and see if anyone knows a possible fix? All I can suggest atm

**Comment by Nicolas-Klatzer** [comment]
I solved the issue?

**Comment by Nicolas-Klatzer** [comment]
It works now

**Comment by Nicolas-Klatzer** [comment]
*How?*

**Comment by Nxybi** [comment]
> _How?_

Put the answer so anyone who gets this error aswell has the solution xD

**Comment by Nicolas-Klatzer** [comment]
Okay, I think this is a very weird error. The timeout decided now to entirely break the system and make the HTTPClient not send any data to Hiven or even receive anything. So basically eliminating the timeout solved it here and apparently, that should also solve it on the Linux side where it initially worked. That's especially weird because it did not before cause any errors but now decided it was time to not work anymore. Oh, well it is solved now so I am happy it works now. Just I gotta write now a timeout handler myself that returns an error if the request failed to not get the response in time so it doesn't stop the flow entirely and lets a request take theoretically forever

**Comment by Nxybi** [comment]
> Okay, I think this is a very weird error. The timeout decided now to entirely break the system and make the HTTPClient not send any data to Hiven or even receive anything. So basically eliminating the timeout solved it here and apparently, that should also solve it on the Linux side where it initially worked. That's especially weird because it did not before cause any errors but now decided it was time to not work anymore. Oh, well it is solved now so I am happy it works now. Just I gotta write now a timeout handler myself that returns an error if the request failed to not get the response in time so it doesn't stop the flow entirely and lets a request take theoretically forever

This is why you're the main dev and not me - you're actually smart

**Comment by Nicolas-Klatzer** [comment]
xD thanks I guess but in those 6 hours I did not expect the timeout that should avoid errors is the error itself

**Comment by Nicolas-Klatzer** [comment]
@Nxybi Do you want to write the timeout handler?

**Comment by Nxybi** [comment]
> @Nxybi Do you want to write the timeout handler?

Again - I am dumb, and have a light headache, you have to

**Comment by Nicolas-Klatzer** [comment]
oki then me gonna do it real quick and also head to bed because me sick

**Comment by Nxybi** [comment]
> oki then me gonna do it real quick and also head to bed because me sick

Makes the both of us

**Comment by Nicolas-Klatzer** [comment]
Resolved by #15
